source/common/upstream/subset_lb.cc

```diff
@@ -35,6 +35,25 @@ SubsetLoadBalancer::SubsetLoadBalancer(
       scale_locality_weight_(subsets.scaleLocalityWeight()) {
   ASSERT(subsets.isEnabled());
 
+  if (fallback_policy_ != envoy::api::v2::Cluster::LbSubsetConfig::NO_FALLBACK) {
```

Comment:
could be just a personal preference, but would rather use deferred initialisation or private helper method not to make constructor code blow up